### ISSUE TITLE
inject-checksums does not work for bundle components

### ISSUE THREAD

**Issue by mboisson** [easybuilders/easybuild-framework#3403]
When working on PR https://github.com/easybuilders/easybuild-easyconfigs/pull/11113 I wanted to use `--inject-checksums` to add them for the bundle components. This did not work. If checksums were absent, nothing was added. If checksums were present, they would be detected and EB would complain about that, but when using `--force`, they would not get over-written.

**Comment by akesandgren** [comment]
This is annoying, we should try to fix this...

**Comment by boegel** [comment]
I agree. The problem is that strictly speaking framework doesn't know about specific easyblocks like `Bundle` (and `components` is an easyblock-specific eaysconfig parameter).

So the logic to inject checksums for bundle components should be implemented in the `Bundle` easyblock itself, in a dedicated method, and framework should call out to that method when it's injecting checksums...
